frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java

```diff
@@ -0,0 +1,182 @@
+      slideOutBranch(branchToSlideOutName);
+      graphTable.queueRepositoryUpdateAndModelRefresh();
+      VcsNotifier.getInstance(project).notifySuccess(/* displayId */ null,
+          /* title */ "",
+          getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-success.of-current.HTML")
+              .fmt(branchToSlideOutName));
+
+    } else {
+      val root = gitRepository.getRoot();
+      val shouldDelete = getDeleteLocalBranchOnSlideOutGitConfigValue(root);
+      if (shouldDelete == null) {
+        ModalityUiUtil.invokeLaterIfNeeded(ModalityState.NON_MODAL, () -> suggestBranchDeletion(branchToSlideOutName));
+      } else {
+        handleBranchDeletionDecision(branchToSlideOutName, shouldDelete);
+      }
+    }
+  }
+
+  @UIEffect
+  private void suggestBranchDeletion(String branchName) {
+    val slideOutOptions = new DeleteBranchOnSlideOutSuggestionDialog(project, branchName).showAndGetSlideOutOptions();
+
+    new Task.Backgroundable(project, getString("action.GitMachete.BaseSlideOutAction.task.title")) {
+      @Override
+      @UIThreadUnsafe
+      public void run(ProgressIndicator indicator) {
+        if (slideOutOptions != null) {
+          handleBranchDeletionDecision(branchName, slideOutOptions.shouldDelete());
+
+          if (slideOutOptions.shouldRemember()) {
+            val value = String.valueOf(slideOutOptions.shouldDelete());
+            setDeleteLocalBranchOnSlideOutGitConfigValue(gitRepository.getRoot(), value);
+          }
+        } else {
+          val title = getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-info.canceled");
+          val message = getString(
+              "action.GitMachete.BaseSlideOutAction.notification.message.slide-out-info.canceled.HTML")
+                  .fmt(branchName);
+          VcsNotifier.getInstance(project).notifyInfo(/* displayId */ null, title, message);
+        }
+      }
+    }.queue();
+  }
+
+  @UIThreadUnsafe
+  private void handleBranchDeletionDecision(String branchName, boolean shouldDelete) {
+    slideOutBranch(branchName);
+    if (shouldDelete) {
+      GitBrancher.getInstance(project).deleteBranch(branchName, Collections.singletonList(gitRepository));
+      VcsNotifier.getInstance(project).notifySuccess(/* displayId */ null,
+          /* title */ "",
+          getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-success.with-delete.HTML")
+              .fmt(branchName));
+      return;
+    } else {
+      VcsNotifier.getInstance(project).notifySuccess(/* displayId */ null,
+          /* title */ "",
+          getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-success.without-delete.HTML")
+              .fmt(branchName));
+    }
+    if (graphTable != null) {
+      graphTable.queueRepositoryUpdateAndModelRefresh();
+    }
+  }
+
+  @UIThreadUnsafe
+  private void slideOutBranch(String branchName) {
```

Comment:
@MaciejG604, pls confirm I did not reproduce https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1333 here XD

Reply:
If there's no `assert ... != null`, then we're rather safe ;)